org.adempiere.base/src/org/compiere/model/PO.java

```diff
@@ -2803,6 +2809,57 @@ public void afterClose(Trx trx) {
 				else if (get_ID() > 0)
 					Adempiere.getThreadPoolExecutor().submit(() -> CacheMgt.get().newRecord(p_info.getTableName(), get_ID()));
 			}
+		} else if (p_info.getTableName().endsWith("_Trl") && CacheMgt.get().hasCache(TRANSLATION_CACHE_TABLE_NAME) && !newRecord) {
```

Comment:
Checking the logic, I think all block of MRecentItems and reset cache must be enclosed within the if (success) block.  There is no need to reset cache if the update was not successful